internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -120,49 +121,88 @@ func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, s c
 	}
 
 	for name, status := range in.DependsOn {
-		// status must be one of < start | complete | success >
 		if !isValidStatus(status) {
 			return errInvalidDependsOnStatus
 		}
-		// essential containers must have 'start' as a status
-		if name == s.workloadName || s.sidecarConfig[name].Essential == nil || aws.BoolValue(s.sidecarConfig[name].Essential) {
-			if status != "start" {
-				return errEssentialContainerStatus
-			}
+		if isEssential(name, s) && !isEssentialStatus(status) {
+			return errEssentialContainerStatus
 		}
 	}
 
 	return nil
 }
 
+func isValidStatus(s string) bool {
+	for _, allowed := range dependsOnStatus {
+		if s == allowed {
+			return true
+		}
+	}
+
+	return false
+}
+
+func isEssential(name string, s convertSidecarOpts) bool {
+	if s.sidecarConfig == nil {
+		return false
+	}
+	if name == s.workloadName || s.sidecarConfig[name].Essential == nil || aws.BoolValue(s.sidecarConfig[name].Essential) {
+		return true
+	}
+
+	return false
+}
+
+func isEssentialStatus(s string) bool {
+	for _, allowed := range essentialContainerStatus {
+		if s == allowed {
+			return true
+		}
+	}
+
+	return false
+}
+
 func validateNoCircularDependencies(s convertSidecarOpts) error {
-	used := make(map[string]bool)
-	path := make(map[string]bool)
 	dependencies, err := buildDependencyGraph(s)
 
-	// don't let nonexistent containers pass
 	if err != nil {
 		return err
 	}
 
-	// don't let circular dependencies happen
-	for node := range dependencies.nodes {
-		if !used[node] && hasCycles(dependencies, used, path, node) {
-			return errCircularDependency
+	if !dependencies.isAcyclic() {
+		if len(dependencies.cycle) == 1 {
+			return fmt.Errorf("circular container dependency chain present: container %s depends on itself", dependencies.cycle[0])
```

Comment:
Can we reword this error so it's not as deeply nested?
```suggestion
			return fmt.Errorf("container %s cannot depend on itself", dependencies.cycle[0])
```

Reply:
Adjusted.